lib/uplink/object.go

```diff
@@ -0,0 +1,37 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package uplink
+
+import (
+	"context"
+	"io"
+	"time"
+
+	"storj.io/storj/pkg/storj"
+)
+
+// ObjectMeta contains metadata about a specific Object
+type ObjectMeta struct {
+	Bucket   string
+	Path     storj.Path
+	IsPrefix bool
+
+	Metadata map[string]string
+
+	Created  time.Time
+	Modified time.Time
+	Expires  time.Time
+
+	Size     int64
+	Checksum string
```

Comment:
Why `string` instead of `[]byte` like in `storj.Stream.Checksum`?

Reply:
good call!